modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/main/resources/META-INF/resources/js/components/scope/Scope.es.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import ClayIcon from '@clayui/icon';
+import getCN from 'classnames';
+import {LearnMessage} from 'frontend-js-components-web';
+import React from 'react';
+
+import {
+	PORTAL_TOOLTIP_TRIGGER_CLASS,
+	SCOPE_TYPES,
+} from '../../utils/constants.es';
+import SelectScope from './SelectScope.es';
+
+export default function Scope({
+	disabled,
+	initialGroupExternalReferenceCode,
+	initialSxpBlueprintExternalReferenceCode,
+	onChange,
+	onResultsUpdate,
+	selectedType,
+	touched,
+}) {
+	const _handleScopeTypeChange = (type) =>
+		onChange({
+			touched: false,
+			type,
+			value: '',
+		});
+
+	const _renderRadioButton = (value, helptext, label) => (
```

Comment:
Instead of a function `_render*`, we should be following react patterns and make this a component.